### DISCUSSION TITLE
Remote Cache with AzuraRelay or AzuraCast

### DISCUSSION THREAD

**Discussion by fabneto** [AzuraCast/AzuraCast#6486] [General]
We are working in a broadcast project with a lot of distributed address spread around the country. And some of the address have a poor internet connection where from time to time they are totally offline.

What is the best option to deploy the broadcast signal in a way the if the internet on remote end drops, they could continue listen cached information? 
Is Azurarelay os AzuraCast capable of doing it but with a central management point ??

Thanks!

**Comment by BusterNeece** [comment]
HLS streaming is a fair bit more fault-tolerant of outages than regular streaming MP3s, since rather than having to maintain one steady connection, HLS involves a ton of little requests that piece together into the audio stream. It might be worth looking into for this scenario.

AzuraCast + AzuraRelay will give you geographic diversity of your stream, so users can connect to relays that are closer (and that can handle more listeners than just the main installation itself), but if the main installation goes down, the relays follow suit quickly.

We do support a rudimentary high-availability setup by pointing two AzuraCast instances at the same databases; the two systems will automatically negotiate which one handles synchronized tasks and other under-the-hood work, and if one fails, the other will take over.
